contents/scalability/scalable-development.md

```diff
@@ -33,9 +33,29 @@ There are a few common bottlenecks that can be avoided such as database querying
 
 There are many program profilers available that you can use to find the bottlenecks in your server. By finding out where the server spends most computational time in, you can potentially save huge amounts of time.
 
-Performance bottlenecks ([resource](https://www.apicasystem.com/blog/5-common-performance-bottlenecks/))
+Performance bottlenecks ([resource](https://www.apicasystem.com/blog/5-common-performance-bottlenecks))
 
-Non-Relational Database ([resource](http://www.jamesserra.com/archive/2015/08/relational-databases-vs-non-relational-databases/)) ([resource](https://www.pluralsight.com/blog/software-development/relational-non-relational-databases))
+Non-Relational Database ([resource](http://www.jamesserra.com/archive/2015/08/relational-databases-vs-non-relational-databases)) ([resource](https://www.pluralsight.com/blog/software-development/relational-non-relational-databases))
+
+### Caching
+When more and more data is fetched from the database, the response becomes slower. This is where caching comes in. A cache is a key-value store that resides between the application and the database. By retrieving data from the cache instead of the database in the back-end server, the response time reduces greatly.
```

Comment:
Isn't clear why the response becomes slower as more and more data is fetched from database. Also, perhaps can be more explicit on where the cache is being stored.

Reply:
Consider replacing

> Whenever more data is fetched from a database, it's response becomes slower.

Further, it would be good to explain the reason why response is slower, and therefore how caching can come in and help solve the issue